velox/dwio/parquet/common/BloomFilter.cpp

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Adapted from Apache Arrow.
+
+#include "BloomFilter.h"
+#include "XxHasher.h"
+#include "velox/dwio/parquet/thrift/ThriftTransport.h"
+
+#include <thrift/protocol/TCompactProtocol.h>
+#include <thrift/transport/TBufferTransports.h>
+
+#include <cstdint>
+#include <cstring>
+#include <memory>
+
+namespace facebook::velox::parquet {
+
+constexpr uint32_t BlockSplitBloomFilter::SALT[kBitsSetPerBlock];
+
+BlockSplitBloomFilter::BlockSplitBloomFilter(memory::MemoryPool* pool)
```

Comment:
The BloomFilter class was moved from writer/arrow, but velox/dwio/parquet/writer/arrow/tests/BloomFilter.cpp is still there. Will you be able to send a separate commit/PR to remove the duplicate code there?

Reply:
Yes. I will remove the ones in writer/arrow once this PR is merged.

Reply:
Can we do this move/cleanup in a different PR? It makes this one unnecessary large and harder to review.